packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -99,16 +100,20 @@ function st(fragment: Element, key: Key, parts?: VStaticPart[]): VStatic {
 
 // [fr]agment node
 function fr(key: Key, children: VNodes, stable: 0 | 1): VFragment {
-    const leading = t('');
-    const trailing = t('');
+    const owner = getVMBeingRendered()!;
+    const useCommentNodes = owner.apiVersion >= APIVersion.V60_248_SPRING_24;
+
+    const leading = useCommentNodes ? co('') : t('');
+    const trailing = useCommentNodes ? co('') : t('');
```

Comment:
For debugging purposes, we could also add a marker in comments in dev mode to indicate the start and end of a fragment.

Should we also update the jest test serializer to get rid of those comments from tests snapshots?

Reply:
Turns out adding the dev-mode content to comment nodes makes a lot of tests more complicated. Take for example the following snippet from one of our tests:

```javascript
const contents = Array.from(childNodes).map((node) => node.textContent);
expect(contents).toEqual(['', '1', '', '', '2', '', '', '3', '']);
```

This would become

```javascript
const contents = Array.from(childNodes).map((node) => node.textContent);
expect(contents).toEqual(['vfragment:start', '1', 'vfragment:end', 'vfragment:start', '2', 'vfragment:end', 'vfragment:start', '3', 'vfragment:end']);
```

But actually, it's more complicated than that. Because in API version < 60, the assertion would have to be the original with empty strings at positions 0, 2, 3, 5, 6, and 8.

For now, I'm inclined to leave the comments nodes with no content.